napari/layers/utils/plane.py

```diff
@@ -0,0 +1,166 @@
+from typing import Tuple
+
+import numpy as np
+from pydantic import validator
+
+from ...utils.events import EventedModel, SelectableEventedList
+from ...utils.geometry import intersect_line_with_plane_3d
+from ...utils.translations import trans
+
+
+class Plane(EventedModel):
+    """Defines a plane in 3D (with optional thickness) and provides utility methods
+    and events to handle it.
+
+    A plane is defined by a position, a normal vector and a thickness value, and can
+    be enabled or disabled.
+
+    Attributes
+    ----------
+    position : 3-tuple
+        A 3D position on the plane, defined in data coordinates.
+    normal : 3-tuple
+        A 3D unit vector normal to the plane, defined in data coordinates.
```

Comment:
Strictly these are probably better thought of as sliced data coordinates as they are 3D and not nD.